swig/cpp/examples/rpc_example.cpp

```diff
@@ -0,0 +1,277 @@
+/**
+ * @file application_changes_example.cpp
+ * @author Mislav Novakovic <mislav.novakovic@sartura.hr>
+ * @brief Example application that uses sysrepo as the configuration datastore. It
+ * prints the changes made in running data store.
+ *
+ * @copyright
+ * Copyright 2016 Deutsche Telekom AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unistd.h>
+#include <iostream>
+#include <signal.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "Session.h"
+
+#define MAX_LEN 100
+
+using namespace std;
+
+void
+print_tree(shared_ptr<Tree> tree)
+{
```

Comment:
While not necessary in this particular example, trees in general may consist of multiple nodes, so for an example application it would be more appropriate to show how to print (and iterate) trees in their entirety, i.e. for example iterate recursively through all nodes in certain order and print some diagram.